enhancements/single-node-production-deployment-approach.md

```diff
@@ -0,0 +1,511 @@
+   from the `cluster-version-operator` to the `cluster-etcd-operator`?
+
+### Test Plan
+
+In order to claim full support for this configuration, we must have CI
+coverage informing the release. An end-to-end job using a single-node
+deployment and running an appropriate subset of the standard OpenShift
+tests will be created and configured to block accepting release images
+unless it passes.
+
+That end-to-end job should also be run against pull requests for the
+operators and other components that are most affected by the new
+configuration, such as the etcd and auth operators.
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. Initial proposal
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this
+enhancement:
+
+- Maturity levels
+    - [`alpha`, `beta`, `stable` in upstream Kubernetes][maturity-levels]
+    - `Dev Preview`, `Tech Preview`, `GA` in OpenShift
+- [Deprecation policy][deprecation-policy]
+
+Clearly define what graduation means by either linking to the [API doc definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning),
+or by redefining what graduation means.
+
+In general, we try to use the same stages (alpha, beta, GA), regardless how the functionality is accessed.
+
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+#### Examples
+
+These are generalized examples to consider, in addition to the aforementioned [maturity levels][maturity-levels].
+
+##### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+- Enumerate service level indicators (SLIs), expose SLIs as metrics
+- Write symptoms-based alerts for the component(s)
+
+##### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- Backhaul SLI telemetry
+- Document SLOs for the component
+- Conduct load testing
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+##### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
+- Deprecate the feature
+
+### Upgrade / Downgrade Strategy
+
+In-place upgrades and downgrades will not be supported for this first
+iteration. Upgrades will initially only be achieved by redeploying
+the machine and its workload.
+
+### Version Skew Strategy
+
+With only one node and no in-place upgrade, there will be no
```

Comment:
this is true for dev preview but is it true for all releases? if not i'dexplicitly flag as:

Dev Preview: with only one...

Reply:
We believe that a minimum-viable version of single-node production deployments does not have to include the ability to upgrade. That won't be useful to all users, but it will be useful for a lot of cases. I've updated the language around upgrades in several places in this latest draft to reflect the fact that upgrades are not being considered at all as part of this enhancement and that we will use a separate enhancement when the requirement for upgrades is presented.